src/engine/GraphStoreProtocol.h

```diff
@@ -94,43 +105,108 @@ class GraphStoreProtocol {
                      ad_utility::truncateOperationString(rawRequest.body()));
     return res;
   }
-  FRIEND_TEST(GraphStoreProtocolTest, transformPost);
+  FRIEND_TEST(GraphStoreProtocolTest, transformPostAndTsop);
+
+  // Transform a SPARQL Graph Store Protocol TSOP to an equivalent ParsedQuery
+  // which is an SPARQL Update.
+  CPP_template_2(typename RequestT)(
+      requires ad_utility::httpUtils::HttpRequest<RequestT>) static ParsedQuery
+      transformTsop(const RequestT& rawRequest, const GraphOrDefault& graph) {
+    if (rawRequest.body().empty()) {
+      throw HttpError(boost::beast::http::status::no_content);
+    }
+    auto triples =
+        parseTriples(rawRequest.body(), extractMediatype(rawRequest));
+    auto convertedTriples = convertTriples(graph, std::move(triples));
+    updateClause::GraphUpdate up{{}, std::move(convertedTriples)};
+    ParsedQuery res;
+    res._clause = parsedQuery::UpdateClause{std::move(up)};
+    // Graph store protocol TSOP requests might have a very large body. Limit
+    // the length used for the string representation.
```

Comment:
Is this your custom operations "literal delete" ? Then comment it in all the places where it occurs, and how did you come up with that name?

Reply:
(okay, it is post backwards, but still...